src/security/srv_acl.c

```diff
@@ -506,6 +506,76 @@ get_sec_capas_for_token(Auth__Token *token, struct ownership *ownership,
 	return rc;
 }
 
+static int
+get_sec_origin_for_token(Auth__Token *token, char **machine)
+{
+	struct drpc_alloc	alloc = PROTO_ALLOCATOR_INIT(alloc);
+	int			machine_size = 0;
+	int			rc = 0;
+	char			*mtmp;
+	Auth__Sys		*authsys;
+
+
+	if (token == NULL || machine == NULL) {
+		D_ERROR("NULL input\n");
+		return -DER_INVAL;
+	}
+
+	rc = get_auth_sys_payload(token, &authsys);
+	if (rc != 0)
+		return rc;
+
+	if (authsys->machinename) {
+		machine_size = strnlen(authsys->machinename, MAXHOSTNAMELEN);
```

Comment:
Should be `MAXHOSTNAMELEN+1` to detect the case where it's longer

Reply:
Good idea. This will catch anything over maxhostnamelen and the check for maxhostnamelen below will ensure we don't go over.